src/main/kotlin/it/pagopa/wallet/domain/Wallet.kt

```diff
@@ -0,0 +1,17 @@
+package it.pagopa.wallet.domain
+
+/**
+ * A wallet.
+ *
+ * A wallet is a collection of payment instruments identified by a single wallet id.
+ *
+ * The following assumptions should always hold:
+ * - Wallets are non-empty
+ * - No two wallets share a payment instrument with the same id (i.e. the relation `wallet <->
+ *   paymentInstrument` is 1:n)
```

Comment:
Add throws IllegalArgumentException declaration into javadoc in case of empty paymentInstrument